src/main/java/seedu/mark/logic/commands/commandresult/TabCommandResult.java

```diff
@@ -0,0 +1,34 @@
+package seedu.mark.logic.commands.commandresult;
+
+/**
+ * Represents the result of a tab command execution.
+ */
+public class TabCommandResult extends CommandResult {
+
+    private final boolean switchViewDashboard;
+    private final boolean switchViewOnline;
+    private final boolean switchViewOffline;
+
+    public TabCommandResult(String feedbackToUser, boolean dashboard, boolean online, boolean offline) {
```

Comment:
Hmm having so many boolean values in this class seems a bit strange. Can `TabCommandResult` make use of the `Tab` enum too?

Reply:
Hmm okay, then I'll get `MainWindow` to handle the various `Tab`s. Hopefully it'll be neater

Reply:
Feels weird to let `CommandResult` to have a `getTab()` too though
I'll make the change first, then we can judge later